### PR TITLE
Assets copy-paste

### LINKED ISSUE #1325: When copying instances between projects, references to Assets and Pages are broken
Depends on: #1335

- [ ] Create a corresponding Asses for `Image.src`
- [ ] Create a corresponding Asset for fonts (`fontFamily`)
- [ ] Handle `Link.href` somehow (see what Webflow does)

The list above are just examples. Need to handle all references to Assets or Pages in props or style.

https://discord.com/channels/955905230107738152/1090935034657120286

### PR DESCRIPTION
**Pull request by rpominov**
## Description

1. Closes #1325
2. Assets used in styles are copied
3. Assets used in props are copied

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
- [ ] hi @TrySound, I need you to do
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/shared/copy-paste/plugin-instance.ts [hunk 1]
```diff
@@ -65,12 +70,92 @@ const findTreeStyleSourceIds = (
...
+      ) {
+        continue;
+      }
+      value satisfies never;
+    }
+  };
+
+  processValues(style.map(({ value }) => value));
+
+  for (const [, asset] of assetsStore.get()) {
```
Comment (TrySound) [inline apps/builder/app/shared/copy-paste/plugin-instance.ts:117]: `const asset of assetsStore.get().values()`

File: apps/builder/app/shared/copy-paste/plugin-instance.ts [hunk 2]
```diff
@@ -164,15 +265,25 @@ export const onPaste = (clipboardData: string) => {
...
       styleSourceSelections,
-      styles
+      styles,
+      assets
     ) => {
+      // if pasting to a different project, copy assets.
+      // use the same ids so the references are preserved.
+      if (data.projectId !== project.id) {
+        for (const asset of data.assets) {
+          assets.set(asset.id, { ...asset, projectId: project.id });
```
Comment (TrySound) [inline apps/builder/app/shared/copy-paste/plugin-instance.ts:283]: Or we can always iterate and check if id exists already before writing

Comment (rpominov) [reply apps/builder/app/shared/copy-paste/plugin-instance.ts:283]: Is there a reason to choose one over the other?

**Review by TrySound — APPROVED** [review]

File: apps/builder/app/shared/copy-paste/plugin-instance.ts (hunk 2 above)
Comment (TrySound) [reply apps/builder/app/shared/copy-paste/plugin-instance.ts:283]: Everything else works this way. And here we additionally need to store project id in clipboard.

**Pull request merged**